### ISSUE TITLE
Add python3.10 to the setup.py

### ISSUE THREAD

**Issue by ruochiz** [open2c/cooler#281]
Could you help to add python3.10 to the setup.py? Thanks.

**Comment by nvictus** [comment]
Certainly! Would you be interested in proposing the change as a pull request? Then, I'll merge it.

**Comment by ruochiz** [comment]
Actually, after a closer look at it, I cannot think of a direct fix for this.

1. when doing conda install cooler -c bioconda, all platforms I tried with python 3.10 would fail with this error:
to be incompatible with the existing python installation in your environment:

Specifications:

  - cooler -> python[version='2.7.*|3.5.*|3.6.*|>=2.7,<2.8.0a0|>=3.6,<3.7.0a0|>=3.7,<3.8.0a0|>=3.5,<3.6.0a0|3.4.*']

Your python: python=3.10

2. when doing pip install cooler

The error occurs at install cytoolz on windows and M1 Macbook pro, which could possibly related to this issue: https://github.com/pytoolz/cytoolz/issues/163

On linux machine, it can be installed without any error.

3. when doing git clone + python setup.py
same error as above

**Comment by nvictus** [comment]
Thanks for looking into this. 

1. I am also seeing the issue with conda installing cooler on a M1 Macbook Air on Python 3.10. I suspect that py3.10 wasn't supported on bioconda during the last recipe build so a new bioconda build needs to be triggered. I'll look into it.

2. This one is odd. I have no problems pip installing cooler on my M1, but I am using an Intel x86 conda environment on rosetta emulation. If some dependency like cytoolz is failing to build on M1, you could try installing it separately using conda and then retry pip installing cooler.

Update: I tried switching to arm64 conda and it wasn't cytoolz but scipy that failed to build. As above, I just used conda to install scipy and retried `pip install cooler` and it worked. Could you give it a try?

**Comment by ruochiz** [comment]
I tried the conda install cytoolz. The problem I encounter now is that the conda installs cytoolz=0.11.0 for python 3.10, I tried to force it to install cytoolz=0.10.1, conda won't solve the environment somehow.

(py3) ➜  cooler git:(master) ✗ conda install cytoolz=0.10
Collecting package metadata (current_repodata.json): done
Solving environment: failed with current_repodata.json, will retry with next repodata source.
Collecting package metadata (repodata.json): done
Solving environment: failed
Initial quick solve with frozen env failed.  Unfreezing env and trying again.
Solving environment: failed

UnsatisfiableError: The following specifications were found to be incompatible with each other:

Package pip conflicts for:
cytoolz=0.10 -> toolz[version='>=0.10.0'] -> python[version='>=3.5'] -> pip
python=3.10 -> pip
Package setuptools conflicts for:
cytoolz=0.10 -> toolz[version='>=0.10.0'] -> python[version='>=3.5'] -> pip -> wheel -> setuptools
python=3.10 -> pip -> wheel -> setuptools


On the other hand, pip install cooler requires cytoolz<0.11 as stated in requirements.txt, so it falls back to installing cytoolz=0.10.1 via pip wheel, and failed. 

The workaround I found is to change the requirements.txt to release the cytoolz<0.11, and then install via python setup.py install.
